htlcswitch/switch.go

```diff
@@ -1540,8 +1552,17 @@ func (s *Switch) htlcForwarder() {
 	s.cfg.FwdEventTicker.Resume()
 	defer s.cfg.FwdEventTicker.Stop()
 
+	defer s.cfg.AckEventTicker.Stop()
+
 out:
 	for {
+
+		// If the set of pending settle/fail entries is non-zero,
+		// reinstate the ack ticker so we can batch ack them.
+		if len(s.pendingSettleFails) > 0 {
```

Comment:
is it possible to move this statement after `handlePacketForward` in the `htlcPlex` case? should be fine since resolutions will never have a destref

Reply:
Do you mean here:
https://github.com/lightningnetwork/lnd/blob/9f065dccb1e1bee4c468d27098f41371695a7450/htlcswitch/switch.go#L1634-L1635
because I checked and couldn't find any resolution messages passing through that channel.  If we moved it, then the ticker would only wake once a new htlcPacket came across the channel after there were pendingSettleFails which seems fine but I prefer the current approach since it's more straightforward when the ticker will wake up.

Reply:
correct, the resolutions are processed in a different case that calls `handlePacketForward`, but they shouldn't have any destrefs and so don't influence the set of things that need to be acked. the current approach is fine by me, it is more conservative but certainly correct :)